### DISCUSSION TITLE
typescript-react-query failing to parse process.env

### DISCUSSION THREAD

**Discussion by jonathanstanley** [dotansimha/graphql-code-generator#5443] [General] [answered]
**Description**
`@graphql-codegen/typescript-react-query` [documentation](https://graphql-code-generator.com/docs/plugins/typescript-react-query) indicates `process.env` or other variables may be used within the `codegen.yml`

this works for the fetcher `endpoint`, but fails with `fetchParams`

**To Reproduce**
[See codesandbox](https://codesandbox.io/s/graphql-codegen-react-query-env-be8w1?file=/types.ts:171-477) for reproduced behavior

**Expected behavior**
should produce:
```ts
const res = await fetch(process.env.ENDPOINT as string, {
      ...
      headers: {apikey:process.env.APIKEY},
      ...
    });
```

instead produces (note the "" in the headers JSON)
```ts
const res = await fetch(process.env.ENDPOINT as string, {
      ...
      headers: {"apikey":"process.env.APIKEY"},
      ...
    });
```

**Additional context**
The failure is caused by  `JSON.stringify(...)` which produces JSON rather than javascript object literal notation.  [See the source code here](https://github.com/dotansimha/graphql-code-generator/blob/master/packages/plugins/typescript/react-query/src/fetcher-fetch-hardcoded.ts#L28).

It is important that generated/committed code should not include API keys / sensitive information for security.  

**Posible solutions**

1. clarify documentation that `process.env` may only be used for the endpoint. 
2. modify `getFetchParams` to allow variables, such as by returning a string representation of Javascript Object Literal Notation (ie drop the double-quotes where appropriate)

The first would be fast, but not great for security.  I haven't thought of a great way to do the latter, but may be willing to submit an issue or PR based on feedback.

**Comment by dotansimha** [comment, marked as answer]
@jonathanstanley It seems like an issue, please report this under `Issues` section with a reproduction and we'll look into that. Thank you!
